FrameworkSystem/private/monitoring/MonitoringCatalog.py

```diff
@@ -11,368 +11,374 @@
 from DIRAC.FrameworkSystem.private.monitoring.Activity import Activity
 from DIRAC.Core.Utilities import Time
 
-class MonitoringCatalog( object ):
 
-  def __init__( self, dataPath ):
+class MonitoringCatalog(object):
+
+  def __init__(self, dataPath):
     """
     Initialize monitoring catalog
     """
     self.dbConn = False
     self.dataPath = dataPath
-    self.log = gLogger.getSubLogger( "ActivityCatalog" )
+    self.log = gLogger.getSubLogger("ActivityCatalog")
     self.createSchema()
 
-  def __connect( self ):
+  def __connect(self):
     """
     Connect to database
     """
     if not self.dbConn:
       dbPath = "%s/monitoring.db" % self.dataPath
-      self.dbConn = sqlite3.connect( dbPath, isolation_level = None )
+      self.dbConn = sqlite3.connect(dbPath, isolation_level=None)
 
-  def __dbExecute( self, query, values = False ):
+  def __dbExecute(self, query, values=False):
     """
     Execute a sql statement
     """
-    cursor = self.dbConn.cursor() #pylint: disable=no-member
-    self.log.debug( "Executing %s" % query )
+    cursor = self.dbConn.cursor()  # pylint: disable=no-member
+    self.log.debug("Executing %s" % query)
     executed = False
-    while not executed:
+    retry = 0
+    while not executed and retry < 10:
+      retry += 1
       try:
         if values:
-          cursor.execute( query, values )
+          cursor.execute(query, values)
         else:
-          cursor.execute( query )
+          cursor.execute(query)
         executed = True
-      except:
-        time.sleep( 0.01 )
+      except BaseException as e:
+        self.log.exception("Exception executing statement", "query: %s, values: %s" % (query, values), lException=e)
```

Comment:
`Exception` and you don't need to use `lException=e`, it will pick it up anyway
```suggestion
      except Exception:
        self.log.exception("Exception executing statement", "query: %s, values: %s" % (query, values)
```

Reply:
Please apply this suggestion and then I approve the PR.

Reply:
done